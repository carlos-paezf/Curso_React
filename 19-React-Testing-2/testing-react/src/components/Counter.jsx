import { useCounter } from '../hooks/useCounter'
import ButtonCounter from './ButtonCounter'

const Counter = () => {

    const { state, handleModifyCounter } = useCounter()

    return (
        <>
            <h2 data-testid="counter" role="counter">Counter: {state}</h2>

            <ButtonCounter name='Aumentar +1' action={handleModifyCounter} value={1} aria='aumentar' />
            <ButtonCounter name='Disminuir +1' action={handleModifyCounter} value={-1} aria='disminuir' />
            <ButtonCounter name='Resetear Contador' action={handleModifyCounter} value={-state} aria='reset' />
        </>
    )
}

export default Counter
