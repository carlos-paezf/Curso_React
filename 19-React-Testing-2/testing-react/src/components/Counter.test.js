import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Counter from './Counter'

describe('<Counter />', () => {

    beforeEach(() =>{
        render(<Counter />)
    })

    it('Primer Snapshots', () => {
        expect(screen.getByRole('counter')).toMatchSnapshot()
    })
    
})

