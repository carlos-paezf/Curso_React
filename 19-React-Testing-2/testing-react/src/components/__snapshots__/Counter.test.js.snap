// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> Primer Snapshots 1`] = `
<h2
  data-testid="counter"
  role="counter"
>
  Counter: 
  0
</h2>
`;
